src/Files.App/Data/Factories/ContentPageContextFlyoutFactory.cs

```diff
@@ -405,7 +405,7 @@ private static bool Check(ContextMenuFlyoutItemViewModel item, CurrentInstanceVi
 				new ContextMenuFlyoutItemViewModel()
 				{
 					Text = "BaseLayoutItemContextFlyoutSetAs/Text".GetLocalizedResource(),
-					ShowItem = itemsSelected && (selectedItemsPropertiesViewModel?.IsSelectedItemImage ?? false),
+					ShowItem = itemsSelected && (selectedItemsPropertiesViewModel?.CanSelectedItemBeManipulated ?? false),
```

Comment:
```suggestion
					ShowItem = itemsSelected && (selectedItemsPropertiesViewModel?.IsCompatibleToSetAsWindowsWallpaper?? false),
```

Can you update the other locations as well?

Reply:
Done